### PR TITLE
[ie/ctvnews] Fix extractor

### LINKED ISSUE #8689: [CTVNews] Does not find video on page
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

canada

### Provide a description that is worded well enough to be understood

URL: https://www.ctvnews.ca/business/respondents-to-bank-of-canada-questionnaire-largely-oppose-creating-a-digital-loonie-1.6665797

Downloads a play list but finds zero items (videos)
No actual error,  just no resulting video.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
>yt-dlp.py -vU "https://www.ctvnews.ca/business/respondents-to-bank-of-canada-questionnaire-largely-oppose-creating-a-digital-loonie-1.6665797 "
[debug] Command-line config: ['-vU', 'https://www.ctvnews.ca/business/respondents-to-bank-of-canada-questionnaire-largely-oppose-creating-a-digital-loonie-1.6665797 ']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2023.11.29.232714 from yt-dlp/yt-dlp-nightly-builds [6a9c7a2b5] (zip)
[debug] Python 3.10.5 (CPython AMD64 64bit) - Windows-10-10.0.14393-SP0 (OpenSSL 1.1.1n  15 Mar 2022)
[debug] exe versions: ffmpeg 6.1-full_build-www.gyan.dev (setts), ffprobe 4.3.2-2021-02-02-full_build-www.gyan.dev, rtmpdump 2.4
[debug] Optional libraries: sqlite3-3.37.2
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1792 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2023.11.29.232714 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2023.11.29.232714 from yt-dlp/yt-dlp-nightly-builds)
[CTVNews] Extracting URL: https://www.ctvnews.ca/business/respondents-to-bank-of-canada-questionnaire-largely-oppose-creating-a-digital-loonie-1.6665797
[CTVNews] 1.6665797: Downloading webpage
[CTVNews] 1.6665797: Downloading webpage
[download] Downloading playlist: 1.6665797
[CTVNews] Playlist 1.6665797: Downloading 0 items
[download] Finished downloading playlist: 1.6665797
```

### PR DESCRIPTION
**Pull request by jshumphrey**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixes #8689 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [X] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/ctvnews.py [hunk 1]
```diff
@@ -40,6 +48,13 @@ class CTVNewsIE(InfoExtractor):
     }, {
         'url': 'http://vancouverisland.ctvnews.ca/video?clipId=761241',
         'only_matching': True,
+    }, {
+        'url': 'https://www.ctvnews.ca/business/respondents-to-bank-of-canada-questionnaire-largely-oppose-creating-a-digital-loonie-1.6665797',
+        'info_dict':
+        {
+            'id': '1.6665797',
+        },
```
Comment (bashonly) [inline yt_dlp/extractor/ctvnews.py:56]:
```suggestion
        'info_dict': {
            'id': '1.6665797',
        },
```

File: yt_dlp/extractor/ctvnews.py [hunk 2]
```diff
@@ -67,4 +82,7 @@ def ninecninemedia_url_result(clip_id):
                 if 'getAuthStates("' in webpage:
                     entries = [ninecninemedia_url_result(clip_id) for clip_id in
                                self._search_regex(r'getAuthStates\("([\d+,]+)"', webpage, 'clip ids').split(',')]
+                else:
+                    entries = [ninecninemedia_url_result(clip_id) for clip_id in orderedSet(
+                        re.findall(r'axisId&#34;:&#34;(\d+)', webpage))]
```
Comment (bashonly) [inline yt_dlp/extractor/ctvnews.py:87]:
```suggestion
                    entries = [ninecninemedia_url_result(clip_id)
                                    for clip_id in orderedSet(re.findall(r'axisId&#34;:&#34;(\d+)', webpage))]
```

**Review by bashonly — APPROVED** [review]
just formatting things

**Review by seproDev — CHANGES_REQUESTED** [review]
Imo. the `if page_id.isdigit():` path should be removed entirely as it is not accurate and sometimes a page can contain multiple videos.
The VALID_URL regex should also be fixed since URLs such as `https://www.ctvnews.ca/canada/apparent-taylor-swift-ticket-scam-targets-hundreds-who-claim-to-be-out-300k-1.7113362` currently get the id `300`

**Comment by bashonly** [comment]
`_VALID_URL` / playlist ID extraction fix is in https://github.com/yt-dlp/yt-dlp/pull/8892

**Review by seproDev — APPROVED** [review]

**Pull request merged**